.github/workflows/build.yml

```diff
@@ -10,12 +10,13 @@ defaults:
     shell: bash
 
 jobs:
-  build_check:
+  pre_build:
     name: Check if build is needed
```

Comment:
Consider changing the name to reflect it also retrieves the node version. 

"Check if build is needed & preparation" or something similar